app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -53,11 +60,48 @@ class PlantDetailFragment : Fragment() {
                 plantDetailViewModel.addPlantToGarden()
                 Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
             }
+
+            setHasOptionsMenu(true)
         }
 
         return binding.root
     }
 
+    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
+        inflater?.inflate(R.menu.menu_plant_detail, menu)
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        return when (item?.itemId) {
+            R.id.action_share -> {
+                share()
+                return true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun share() {
```

Comment:
It could be a wonder that the "share"  will be packed as an `extension` of `activity` or `fragment` which has title and content as arguments. As learning project for people who want to get some reusable codes, the extension can help more. How do you think?

Reply:
i dont think "share" should tie to an `Activity` or `Fragment`. this is similar to String case where people wanted to extend it. but would it be appropriate? 
i'm not too sure how you do reusable code in kotlin, in java there is always a "Util" class

Reply:
lemme change it to `fun getShareTextIntent(text: String): Intent` for more flexible use case.